sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpan.java

```diff
@@ -35,7 +36,7 @@
 
 /** Implementation for the {@link Span} class that records trace events. */
 @ThreadSafe
-final class SdkSpan implements ReadWriteSpan {
+final class SdkSpan implements ReadWriteSpan, ExtendedSpan {
```

Comment:
Initially I switched ReadWriteSpan to extend ExtendedSpan instead of Span, but that doesn't work because ReadWriteSpan is part of the public API, and we can't have experimental surface area exposed in the stable API. 

We have a long history of doing non-ideal things to be able to incubate things without committing to the stable API. While this is a smell, it will be short lived as it goes away as soon as the spec stabilizes and these methods are promoted to span.

> Immediately makes me question whether or not ExtendedSpan really needs to extend Span.

Any other ideas?

Reply:
> We have a long history of doing non-ideal things to be able to incubate things without committing to the stable API.

👍

Reply:
> > Immediately makes me question whether or not ExtendedSpan really needs to extend Span.
> 
> Any other ideas?

What if it just didn't? It can be its own stand-alone interface, right? And it's only used by the single implementation.

Reply:
Well the idea is that a user casts spans to it:
```
ExtendedSpan span = (ExtendedSpan) tracer.spanBuilder("name").startSpan();
span.setAttribute("key", "value");
span.addLink(spanContext);
```
If ExtenedSpan doesn't extend Span, then its more awkward to use:
```
Span span = tracer.spanBuilder("name").startSpan();
span.setAttribute("key", "value");
// Cast span to ExtendedSpan just to access #addLink
((ExtendedSpan) span).addLink(spanContext);
```

Reply:
Yeah ok, I get it. Thanks for clarifying.